test/unit/adversarial/helper.py

```diff
@@ -23,6 +23,7 @@
 from fairlearn.adversarial._tensorflow_engine import TensorflowEngine
 
 model_class = type("Model", (object,), {})
+Model = type("Model", (model_class,), {})
```

Comment:
this whole thing is weird. Talked with @TamaraAtanasoska , and it seems nicer to have a separate CI, one for TF and one for pytorch, instead of mocking these here.

Reply:
I have just reverted the changes for this PR that just address the pickling issues, and I'll make a separate PR with the changes.